### DISCUSSION TITLE
How can I configure custom content in the aside-container within Theme.Layout?

### DISCUSSION THREAD

**Discussion by xi-mad** [web-infra-dev/rspress#1637] [Q&A] [answered]
How can I configure custom content in the aside-container within Theme.Layout? I successfully replaced the content of the main section using the wrapper. Here is my code:
```
      <Theme.Layout
        components={{
          // replace the content
          wrapper: ({ children }: { children: any }) => <CustomContent postsByYear={postsByYear}>{children}</CustomContent>,

          // replace the sidebar ??
          sidebar: ({ children }: { children: any }) => <div>右侧{children}</div>,
        }}
      />
```

<img width="1225" alt="image" src="https://github.com/user-attachments/assets/34c2fe38-f65f-4c68-b908-e6b73ed23118">

**Comment by Timeless0911** [comment, marked as answer]
The `components` props you current using is usually used to customize HTML elements like `H1`, `p`, ...etc.

If you customize the content, TOC data cannot be generated during compilation, so the `Aside` is empty. You can simply use `beforeOutline` hook in `<Theme.Layout />` or you should [Redevelop a custom theme](https://rspress.dev/guide/advanced/custom-theme#redevelop-a-custom-theme) since `Aside` component is a built-in component in Layout.

**Comment by xi-mad** [comment]
@Timeless0911 Thank you so much for providing me with another approach. I have successfully achieved the desired effect using `beforeOutline`, and I hope this solution can help more people. Thanks again!

<img width="1020" alt="image" src="https://github.com/user-attachments/assets/1aa5fe52-fb36-4828-9cd4-a5d544f7fbe2">
